src/Simulation/Core/Operations/Operation.cs

```diff
@@ -78,13 +78,18 @@ public Operation(IOperationFactory m) : base(m)
         public ControlledOperation<I, O> Controlled => _controlled.Value;
 
         /// <inheritdoc/>
-        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args) =>
-            new RuntimeMetadata()
+        public override RuntimeMetadata? GetRuntimeMetadata(IApplyData args)
+        {
+            var targets = args.GetQubits() ?? new List<Qubit>();
+            // Remove duplicate qubits
+            targets = new HashSet<Qubit>(targets).ToList();
+            return new RuntimeMetadata()
             {
                 Label = ((ICallable)this).Name,
                 FormattedNonQubitArgs = args.GetNonQubitArgumentsAsString() ?? "",
-                Targets = args.GetQubits() ?? new List<Qubit>(),
+                Targets = targets,
```

Comment:
You may be able to use the `Enumerable.Distinct` extension method here instead of manually constructing a `HashSet`.

Reply:
Oh nice! I'll give it a shot, thanks!